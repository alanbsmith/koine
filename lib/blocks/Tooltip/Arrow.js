import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  bottom: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: 0;
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-top-color: ${props.theme.colors.base.chrome900};
    bottom: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  left: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: 0;
    border-right-color: ${props.theme.colors.base.chrome900};
    left: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  right: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: 0;
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-left-color: ${props.theme.colors.base.chrome900};
    right: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  top: (props) => `
    border-top-width: 0;
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-color: ${props.theme.colors.base.chrome900};
    top: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
};

const Arrow = styled.div`
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Arrow;
