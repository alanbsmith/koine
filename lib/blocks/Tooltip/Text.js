import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';
import gridScale from '../../utils/gridScale';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const Text = styled.span`
  border-radius: ${props => props.theme.dimensions.borderRadius};
  color: ${props => props.theme.colors.base.chrome000};
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Text;
