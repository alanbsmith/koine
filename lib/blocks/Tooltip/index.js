import styled from 'styled-components';
import gridScale from '../../utils/gridScale';

import Arrow from './Arrow';
import Text from './Text';

const Tooltip = styled.div`
  background-color: ${props => props.theme.colors.base.chrome900};
  border-radius: ${props => props.theme.dimensions.borderRadius};
  box-shadow: 0 3px 5px 0 ${props => props.theme.colors.base.shadow};
  display: flex;
  align-items: center;
  justify-content: center;
  padding: ${props => `${gridScale(props, 1)} ${gridScale(props, 2)}`};
  position: relative;
`;

Tooltip.Arrow = Arrow;
Tooltip.Text = Text;

export default Tooltip;
