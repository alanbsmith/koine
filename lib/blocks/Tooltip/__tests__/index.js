import React from 'react';
import Tooltip from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Tooltip Block', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Tooltip>
        <Tooltip.Arrow modifiers={['bottom']} />
        <Tooltip.Text>
          tooltip text
        </Tooltip.Text>
      </Tooltip>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
