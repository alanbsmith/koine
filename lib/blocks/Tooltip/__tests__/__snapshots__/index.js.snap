// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Block renders correctly 1`] = `
.c0 {
  background-color: #191919;
  border-radius: 2px;
  box-shadow: 0 3px 5px 0 rgba(100,100,100,0.4);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  position: relative;
}

.c1 {
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  border-top-width: 0.5rem;
  border-right-width: 0.5rem;
  border-bottom-width: 0;
  border-left-width: 0.5rem;
  border-top-color: #191919;
  bottom: -0.4375rem;
}

.c2 {
  border-radius: 2px;
  color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <span
    className="c2"
  >
    tooltip text
  </span>
</div>
`;
